nyaa/api_handler.py

```diff
@@ -186,3 +190,69 @@ def v2_api_upload():
         # Map errors back from form fields into the api keys
         mapped_errors = {UPLOAD_API_FORM_KEYMAP.get(k, k): v for k, v in upload_form.errors.items()}
         return flask.jsonify({'errors': mapped_errors}), 400
+
+
+# ####################################### INFO #######################################
+ID_PATTERN = '^[1-9][0-9]*$'
+INFO_HASH_PATTERN = '^[0-9a-fA-F]{40}$'  # INFO_HASH as string
+
+
+@api_blueprint.route('/v2/info/<torrent_id_or_hash>', methods=['GET'])
+@basic_auth_user
+@api_require_user
+def v2_api_info(torrent_id_or_hash):
+    torrent_id_or_hash = torrent_id_or_hash.lower().strip()
+
+    matchID = re.match(ID_PATTERN, torrent_id_or_hash)
+    matchHASH = re.match(INFO_HASH_PATTERN, torrent_id_or_hash)
+
+    torrent = None
+
+    if matchID:
+        torrent = models.Torrent.by_id(int(torrent_id_or_hash))
+    elif matchHASH:
+        # Convert the string representation of a torrent hash back into a binary representation
+        a2b_hash = binascii.unhexlify(torrent_id_or_hash)
+        torrent = models.Torrent.by_info_hash(a2b_hash)
+    else:
+        return flask.jsonify({'errors': ['Query was not a valid id or hash.']}), 400
+
+    viewer = flask.g.user
+
+    if not torrent:
+        return flask.jsonify({'errors': ['Query was not a valid id or hash.']}), 400
+
+    # Only allow admins see deleted torrents
+    if torrent.deleted and not (viewer and viewer.is_superadmin):
+        return flask.jsonify({'errors': ['Query was not a valid id or hash.']}), 400
+
+    submitter = None
+    if not torrent.anonymous and torrent.user:
+        submitter = torrent.user.username
+    if torrent.user and (viewer == torrent.user or viewer.is_moderator):
+        submitter = torrent.user.username
+
+    files = {}
+    if torrent.filelist:
+        files = json.loads(torrent.filelist.filelist_blob.decode('utf-8'))
+
+    # Create a response dict with relevant data
+    torrent_metadata = {
+        'submitter': submitter,
+        'url': flask.url_for('view_torrent', torrent_id=torrent.id, _external=True),
+        'id': torrent.id,
+        'name': torrent.display_name,
+        'hash_b32': torrent.info_hash_as_b32, #as used in magnet uri
+        'hash_hex': torrent.info_hash.hex(), #as shown in torrent client
+        'magnet': torrent.magnet_uri,
+        'main_category': torrent.main_category.name,
+        'main_category_id': torrent.main_category.id,
+        'sub_category': torrent.sub_category.name,
+        'sub_category_id': torrent.sub_category.id,
+        'information': torrent.information,
+        'description': torrent.description,
+        'stats': {'seeders': torrent.stats.seed_count, 'leechers': torrent.stats.leech_count, 'downloads': torrent.stats.download_count},
+        'filelist': files
```

Comment:
On another look, this could just be `files`, clean and simple - it's not really a list either.